src/com/google/enterprise/adaptor/filenet/ConfigOptions.java

```diff
@@ -96,26 +98,23 @@ public ConfigOptions(AdaptorContext context)
     logger.log(Level.CONFIG, "filenet.objectFactory: {0}", objectFactoryName);
 
     // TODO(jlacey): Replace this with a MessageFormat pattern.
-    String workplaceUrl = config.getValue("filenet.displayUrl");
-    if (workplaceUrl.endsWith("/getContent/")) {
-      workplaceUrl = workplaceUrl.substring(0, workplaceUrl.length() - 1);
-    }
-    if (workplaceUrl.contains("/getContent")
-        && workplaceUrl.endsWith("/getContent")) {
-      workplaceUrl += "?objectStoreName=" + objectStoreName
-          + "&objectType=document&versionStatus=1&vsId=";
-    } else {
-      workplaceUrl += "/getContent?objectStoreName=" + objectStoreName
-          + "&objectType=document&versionStatus=1&vsId=";
-    }
-    displayUrl = workplaceUrl;
-    logger.log(Level.CONFIG, "displayUrl: {0}", displayUrl);
+    String pattern = config.getValue("filenet.displayUrlPattern");
     try {
-      new ValidatedUri(displayUrl + percentEscape(Id.ZERO_ID.toString()))
+      URI uri = new URI(
+          getDisplayUrl(pattern, Id.ZERO_ID, Id.ZERO_ID, objectStoreName));
+      if (!uri.isAbsolute()) {
+        URI ceUri = new URI(contentEngineUrl);
+        pattern = ceUri.getScheme() + "://" + ceUri.getRawAuthority()
```

Comment:
The problem is logging errors (as you point out below). If constructing the URI throws an exception (unlikely as it would be by now),  I wouldn't have the URL string to log (although it will likely be in the cause message).